uint32 simulation_step
uint32 timestamp
uint32 deadline
string id